string processing bug (Windows 10, MSVC)

### Describe the bug

when msvc compile the file, it show "error C2001: 常量中有换行符", tcc has no this bug

### Reproduction Steps

test.v

```v
module main

fn main() {
	name := '李杰'
	println('${name},你好啊')
}
```

v . -prod -keepc -showcc


### Expected Behavior

compile ok

### Current Behavior

```sh
> C compiler cmd: "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.38.33130\bin\Hostx64\x64\cl.exe" "@C:\Users\DDT\AppData\Local\Temp\v_0\test.tmp.c.rsp"
> C compiler response file "C:\Users\DDT\AppData\Local\Temp\v_0\test.tmp.c.rsp":
-w /we4013 /volatile:ms /Fo"C:\Users\DDT\AppData\Local\Temp\v_0\test.tmp.c.obj" /F 16777216 /O2 /MD /DNDEBUG "C:\Users\DDT\AppData\Local\Temp\v_0\test.tmp.c" -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt" -I "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.38.33130\include" -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um" -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared" /DGC_NOT_DLL=1 /DGC_WIN32_THREADS=1 /DGC_THREADS=1 -I "D:\v\v\v\thirdparty\libatomic_ops" -I "D:\v\v\v\thirdparty\libgc\include" "D:\v\v\v/thirdparty/libatomic_ops/atomic_ops.obj" "D:\v\v\v/thirdparty/libgc/gc.obj" kernel32.lib user32.lib advapi32.lib ws2_32.lib dbghelp.lib /link /NOLOGO /OUT:"D:\v\D2\test\test.exe" /LIBPATH:"C:\Program Files (x86)\Windows Kits\10\Lib\10.0.22621.0\ucrt\X64" /LIBPATH:"C:\Program Files (x86)\Windows Kits\10\Lib\10.0.22621.0\um\X64" /LIBPATH:"C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.38.33130\lib\X64" /DEBUG:FULL /INCREMENTAL:NO /OPT:REF /OPT:ICF
用于 x64 的 Microsoft (R) C/C++ 优化编译器 19.38.33133 版
版权所有(C) Microsoft Corporation。保留所有权利。

cl -w /we4013 /volatile:ms /Fo"C:\Users\DDT\AppData\Local\Temp\v_0\test.tmp.c.obj" /F 16777216 /O2 /MD /DNDEBUG "C:\Users\DDT\AppData\Local\Temp\v_0\test.tmp.c" -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt" -I "C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.38.33130\include" -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\um" -I "C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\shared" /DGC_NOT_DLL=1 /DGC_WIN32_THREADS=1 /DGC_THREADS=1 -I "D:\v\v\v\thirdparty\libatomic_ops" -I "D:\v\v\v\thirdparty\libgc\include" "D:\v\v\v/thirdparty/libatomic_ops/atomic_ops.obj" "D:\v\v\v/thirdparty/libgc/gc.obj" kernel32.lib user32.lib advapi32.lib ws2_32.lib dbghelp.lib /link /NOLOGO /OUT:"D:\v\D2\test\test.exe" /LIBPATH:"C:\Program Files (x86)\Windows Kits\10\Lib\10.0.22621.0\ucrt\X64" /LIBPATH:"C:\Program Files (x86)\Windows Kits\10\Lib\10.0.22621.0\um\X64" /LIBPATH:"C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.38.33130\lib\X64" /DEBUG:FULL /INCREMENTAL:NO /OPT:REF /OPT:ICF

test.tmp.c
C:\Users\DDT\AppData\Local\Temp\v_0\test.tmp.c(12848): error C2001: 常量中有换行符
C:\Users\DDT\AppData\Local\Temp\v_0\test.tmp.c(12848): fatal error C1057: 宏扩展中遇到意外的文件结束

builder error: msvc error
```

### Possible Solution

N/A

### Additional Information/Context

```v
module main

fn main() {
	name := '李杰'
	println('${name}，你好啊')
}
```
will pass the compile.


PS:
This is the BUG generated c:
```c
VV_LOCAL_SYMBOL void main__main(void) {
	string name = _SLIT("\346\235\216\346\235\260");
	println( str_intp(2, _MOV((StrIntpData[]){{_SLIT0, /*115 &string*/0xfe10, {.d_s = name}}, {_SLIT(",你好啊"), 0, { .d_c = 0 }}})));
}
```

This is the OK generated c:
```c
VV_LOCAL_SYMBOL void main__main(void) {
	string name = _SLIT("\346\235\216\346\235\260");
	println( str_intp(2, _MOV((StrIntpData[]){{_SLIT0, /*115 &string*/0xfe10, {.d_s = name}}, {_SLIT("，你好啊"), 0, { .d_c = 0 }}})));
}
```
### V version

V full version: V 0.4.3 66d222c

### Environment details (OS name and version, etc.)

V full version: V 0.4.3 66d222c
OS: windows, Microsoft Windows 10 ҵ LTSC v19044 64 λ
Processor: 16 cpus, 64bit, little endian,

getwd: D:\v\D2\test
vexe: D:\v\v\v\v.exe
vexe mtime: 2024-01-08 00:59:37

vroot: OK, value: D:\v\v\v
VMODULES: OK, value: C:\Users\DDT\.vmodules
VTMP: OK, value: C:\Users\DDT\AppData\Local\Temp\v_0

Git version: git version 2.43.0.windows.1
Git vroot status: weekly.2024.01.test2-19-g66d222cc-dirty (4 commit(s) behind V master)
.git/config present: true

CC version: Error: 'cc' is not recognized as an internal or external command,
operable program or batch file.

thirdparty/tcc status: thirdparty-windows-amd64 b99a453d



> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/str_intp.v
+++ b/vlib/v/gen/c/str_intp.v
@@ -265,7 +265,7 @@ fn (mut g Gen) string_inter_literal(node ast.StringInterLiteral) {
 	g.write(' str_intp(${node.vals.len}, ')
 	g.write('_MOV((StrIntpData[]){')
 	for i, val in node.vals {
-		mut escaped_val := util.smart_quote(val, false)
+		mut escaped_val := cescape_nonascii(util.smart_quote(val, false))
 		escaped_val = escaped_val.replace('\0', '\\0')
 
 		if escaped_val.len > 0 {

--- a/vlib/v/tests/string_interpolation_test.v
+++ b/vlib/v/tests/string_interpolation_test.v
@@ -226,3 +226,10 @@ fn test_contains_zero_characters() {
 	assert str.len == 7
 	assert str == 'abc\0abc'
 }
+
+// for issue: 20432
+// non-ascii escape was lost
+fn test_interpo_non_ascii_characters() {
+	hello := '你好'
+	assert '${hello},世界！' == '你好,世界！'
+}

